crates/router/src/connector/nmi/transformers.rs

```diff
@@ -0,0 +1,653 @@
+    ) -> Result<Self, Self::Error> {
+        let security_key: NmiAuthType = (&item.1).try_into()?;
+        let item = item.0;
+        let security_key = security_key.api_key;
+
+        Ok(Self {
+            transaction_type: TransactionType::Capture,
+            security_key,
+            transactionid: item.connector_transaction_id.clone(),
+            amount: Some(utils::convert_to_higher_denomination(
+                item.amount,
+                item.currency,
+            )?),
+        })
+    }
+}
+
+impl
+    TryFrom<
+        types::ResponseRouterData<
+            api::Capture,
+            GenericResponse,
+            types::PaymentsCaptureData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<api::Capture, types::PaymentsCaptureData, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            api::Capture,
+            GenericResponse,
+            types::PaymentsCaptureData,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let status: enums::AttemptStatus = match item.response.response {
+            Response::Approved => enums::AttemptStatus::CaptureInitiated,
+            Response::Declined | Response::Error => enums::AttemptStatus::CaptureFailed,
+        };
+        Ok(Self {
+            status,
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(item.response.transactionid),
+                redirection_data: None,
+                mandate_reference: None,
+                connector_metadata: None,
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct NmiCancelRequest {
+    #[serde(rename = "type")]
+    pub transaction_type: TransactionType,
+    pub security_key: String,
+    pub transactionid: String,
+    pub void_reason: Option<String>,
+}
+
+impl TryFrom<(&types::PaymentsCancelData, ConnectorAuthType)> for NmiCancelRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: (&types::PaymentsCancelData, ConnectorAuthType),
+    ) -> Result<Self, Self::Error> {
+        let security_key: NmiAuthType = (&item.1).try_into()?;
+        let item = item.0;
+        let security_key = security_key.api_key;
+
+        Ok(Self {
+            transaction_type: TransactionType::Void,
+            security_key,
+            transactionid: item.connector_transaction_id.clone(),
+            void_reason: item.cancellation_reason.clone(),
+        })
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub enum Response {
+    #[serde(alias = "1")]
+    Approved,
+    #[serde(alias = "2")]
+    Declined,
+    #[serde(alias = "3")]
+    Error,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct GenericResponse {
```

Comment:
give better naming

Reply:
renamed it to `StandardResponse`. Would that be okay? Or `CommonResponse`?